Tests/YPersistenceTests/Manager/PersistenceMnager+FetchRecordsTests.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  PersistenceMnager+FetchRecordsTests.swift
+//  YPersistence
+//
+//  Created by Sahil Saini on 18/04/23.
+//  Copyright © 2023 Y Media Labs. All rights reserved.
+//
+
+import XCTest
+@testable import YPersistence
+
+final class PersistenceMnagerFetchRecordsTests: PersistenceManagerBaseTests {
+    func test_fetchRecords_deliversCorrectResults() throws {
+        let sut = makeSUT()
+
+        try confirmEmpty()
+        try insertGroceryProducts()
+
+        let allRecords: [ManagedGroceryProduct] = try sut.fetchRecords(
+            predicate: nil,
+            sortDescriptors: nil,
+            context: nil
+        )
+
+        XCTAssertEqual(allRecords.count, 4)
+    }
+
+    func test_fetchRecordsWithContext_deliversCorrectResults() throws {
+        let sut = makeSUT()
+
+        try confirmEmpty()
+        try insertGroceryProducts()
+
+        let context = sut.container.newBackgroundContext()
+
+        let allRecords: [ManagedGroceryProduct] = try sut.fetchRecords(
+            predicate: nil,
+            sortDescriptors: nil,
+            context: context
+        )
+
+        XCTAssertEqual(allRecords.count, 4)
+    }
+}
+
+extension PersistenceMnagerFetchRecordsTests {
+    func makeSUT() -> PersistenceManager {
```

Comment:
You don't need this because the base class creates a manager in `setUp()` and cleans it up in `tearDown()`